aws/resource_aws_budgets_budget.go

```diff
@@ -0,0 +1,431 @@
+	return map[string]*schema.Schema{
+		"account_id": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validateAwsAccountId,
+		},
+		"name": {
+			Type:          schema.TypeString,
+			Optional:      true,
+			ConflictsWith: []string{"name_prefix"},
+		},
+		"name_prefix": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ForceNew: true,
+		},
+		"budget_type": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"limit_amount": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"limit_unit": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"cost_types": {
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"include_credit": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"include_other_subscription": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"include_recurring": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"include_refund": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"include_subscription": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"include_support": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"include_tax": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"include_upfront": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  true,
+					},
+					"use_blended": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  false,
+					},
+				},
+			},
+		},
+		"time_period_start": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"time_period_end": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Default:  "2087-06-15_00:00",
+		},
+		"time_unit": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"cost_filters": {
+			Type:     schema.TypeMap,
+			Optional: true,
+			Computed: true,
+		},
+	}
+}
+
+func resourceAwsBudgetsBudgetCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+	var accountID string
+	if v, ok := d.GetOk("account_id"); ok {
+		accountID = v.(string)
+	} else {
+		accountID = meta.(*AWSClient).accountid
+	}
+
+	budget, err := expandBudgetsBudgetUnmarshal(d)
+	if err != nil {
+		return fmt.Errorf("failed creating budget: %v", err)
+	}
+
+	_, err = client.CreateBudget(&budgets.CreateBudgetInput{
+		AccountId: &accountID,
+		Budget:    budget,
+	})
+	if err != nil {
+		return fmt.Errorf("create budget failed: %v", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", accountID, *budget.BudgetName))
+	return resourceAwsBudgetsBudgetUpdate(d, meta)
```

Comment:
Do we need to call update here? We generally prefer calling read after create.

Reply:
i was trying to align to convention, seems i copied a non-standard resources flow :).. lol `s3_bucket`

makes more sense without the create->update